File: docs/examples/inheritance/ruby/inheritance_external.rb
OSO ||= Oso.new

class User
  attr_accessor :role

  def initialize(role:, treated:)
    @role = role
    @_treated = treated
  end

  def treated(patient)
    @_treated.include?(patient)
  end
end

OSO.register_class(User)

# start-patient-data
class PatientData
  attr_accessor :patient

  def initialize(patient:)
    @patient = patient
  end
end
OSO.register_class(PatientData)

class Lab < PatientData; end
OSO.register_class(Lab)

class Order < PatientData; end
OSO.register_class(Order)

class Test < PatientData; end
OSO.register_class(Test)
# end-patient-data

File: docs/examples/inheritance/python/inheritance_external.py
import oso


class User:
    def __init__(self, role, treated):
        self.role = role
        self._treated = treated

    def treated(self, patient):
        return patient in self._treated


# start-patient-data
class PatientData:
    def __init__(self, patient):
        self.patient = patient


class Lab(PatientData):
    pass


class Order(PatientData):
    pass


class Test(PatientData):
    pass
    # end-patient-data
